### DISCUSSION TITLE
使用 VapourSynth 缩放视频

### DISCUSSION THREAD

**Discussion by couleurm** [Blinue/Magpie#293] [General]
嘿，你知道使用 VapourSynth 使用 README 中提到的缩放过滤器的任何方法（用于快速 upscaling）

/

Hey, do you know any ways to use the scaling filters mentionned in the README using VapourSynth (for a quick upscaling)

**Comment by Blinue** [comment]
Most of Magpie's filters are ported from mpv hooks. The format of them is unique and unlikely to be used in other applications.

**Reply by couleurm** [reply]
libplacebo?

**Reply by Blinue** [reply]
What you want is probably to port the mpv hook to a VapourSynth filter. I'm not familiar with VapourSynth, you may need to look elsewhere for help.

**Reply by couleurm** [reply]
I've already used ffmpeg/VapourSynth to make use of a FSRCNNX GLSL shader. I'll keep an eye on this discussion if someone more familiar with vs shows up

**Reply by hooke007** [reply]
我没理解错的话，他说的vs自带的缩放应该就是传统的的spline/bicubic这些算法。。。
http://vapoursynth.com/doc/functions/video/resize.html
magpie的着色器实现肯定远比vs的软件缩放快得多

还是说他要port这些magpie的hlsl到glsl？

**Reply by dtlnor** [reply]
> 
Most of the (common) filters suitable for video/image scaling are supported by vapoursynth (Lanczos, bicubic, nnedi3, cugan, or any AI-based filter). Check https://vsdb.top/vsrepogui to find your desired one. If there are some scaling filters that you like in magpie but no public resource (vs plugin) for vapoursynth, ask in doom9 if someone wants to port them to a vs plugin or compile it yourself. As "How to support/port a filter that a random software has to vapoursynth" is not related to that random software, it's a "vapoursynth question".

**Reply by dtlnor** [reply]
Use vsgan for any gan model you'd like to use. Here is some [info](https://forum.doom9.org/showthread.php?t=184000) in case anyone wants to take a try on vapoursynth.
